secure/storage/vault/src/lib.rs

```diff
@@ -159,13 +172,174 @@ impl Client {
                     .remove(key)
                     .ok_or_else(|| Error::NotFound(secret.into(), key.into()))?;
                 let created_time = data.metadata.created_time.clone();
-                Ok((value, created_time))
+                let version = data.metadata.version;
+                Ok(ReadResponse::new(created_time, value, version))
             }
             404 => Err(Error::NotFound(secret.into(), key.into())),
             _ => Err(response.into()),
         }
     }
 
+    pub fn create_ed25519_key(&self, name: &str, exportable: bool) -> Result<(), Error> {
+        let response = ureq::post(&format!("{}/v1/transit/keys/{}", self.host, name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
```

Comment:
Nit: do you think it makes sense to pull these timeout values into a default const? There seems to be a lot of them.